apps/transport/lib/transport_web/controllers/resource_controller.ex

```diff
@@ -161,61 +161,56 @@ defmodule TransportWeb.ResourceController do
       end
 
     conn
-    |> assign_base_resource_details(params, resource, validation_details)
+    |> assign_base_resource_details(params, resource, validation, validation_details)
     |> assign(:validator, Transport.Validators.GTFSTransport)
     |> assign(:data_vis, encoded_data_vis(issue_type, validation))
     |> render("gtfs_details.html")
   end
 
-  defp build_gtfs_validation_details(params, resource) do
-    case latest_validation(resource) do
-      %{result: validation_result, metadata: metadata = %DB.ResourceMetadata{}} ->
-        summary = Transport.Validators.GTFSTransport.summary(validation_result)
-        stats = Transport.Validators.GTFSTransport.count_by_severity(validation_result)
-        issues = Transport.Validators.GTFSTransport.get_issues(validation_result, params)
+  defp build_gtfs_validation_details(nil, _params), do: {nil, nil, nil, [], []}
```

Comment:
Nice le pattern matching pour éviter le `case` qui était présent avant, c'est plus facile à suivre !

Reply:
L'usage "par pattern matching" est le plus idiomatique (souvent recommandé même dans les advents of code etc), ça se retrouve dans la codebase d'Elixir directement.

TIL que l'approche `case` est plus idiomatique en Haskell, on dirait (https://elixirforum.com/t/what-is-the-best-pattern-matching-functions-arguments-or-in-functions-body/59647)!

Reply:
🤣